### PR TITLE
Add type definition for typescript

### PR DESCRIPTION
**Pull request by tjoskar**
Added typescript types, as discussed here: #68 

Questions:

1. `tags` is mention under [setup](https://github.com/logdna/nodejs#setup) but it is not [documented](https://github.com/logdna/nodejs#options-1)? But I guess it should be included in the types?

2. What level does logdna support? According to the [docs](https://github.com/logdna/nodejs#level-1) `YourCustomLevel` is supported so I guess any string is a valid level? But when I look at [this line](https://github.com/logdna/nodejs/blob/1d2cc3a50373791b4844656d93019cd2d617cb44/lib/logger.js#L3) 8 different levels are supported? But only 6 of them are listed in the config. What should the type of `level` be?

3. `cleanUpSIGTERM` and `cleanUpSIGINT` are listed in the [docs](https://github.com/logdna/nodejs#cleanupsigterm) but not in the code(?) should it be removed from the README file?

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+declare module "logdna" {
+  type Level = "Debug" | "Trace" | "Info" | "Warn" | "Error" | "Fatal";
```
Comment (tjoskar) [inline types.d.ts]: This should maybe be a `string`? If any custom level is supported.

File: types.d.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  export function createLogger(
+    key: string,
+    options?: ConstructorOptions
+  ): LogDna;
+  
+  /**
+   * Flushes all existing loggers that are instantiated by createLogger.
+   */
+  export function flushAll(): void;
```
Comment (renarsvilnis) [inline types.d.ts]: Missing optional callback parameter.

Should be: 
```typescript
export function flushAll(callback?: (errMessage?: string) => void): void;
```

File: types.d.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  /**
+   * Flushes all existing loggers that are instantiated by createLogger, and then removes
+   * references to them. Should only be called when you are finished logging.
+   */
+  export function cleanUpAll(): void;
+}
```
Comment (renarsvilnis) [inline types.d.ts:107]: Missing optional callback parameter.

Should be: 
```typescript
export function cleanUpAll(callback?: (errMessage?: string) => void): void; 
```

**Review by renarsvilnis — CHANGES_REQUESTED** [review]
Found that `flushAll()` and `cleanUpAll()` are missing optional callback functions.

Provided the fix, but found it a bit weird that if there is an error while flushing 🚽, the "error" is returned as a string instead of `new Error()` instance.

**Comment by tjoskar** [comment]
Thanks @renarsvilnis and great points!

**Comment by SpainTrain** [comment]
Super excited for this PR.  Is this likely to make it through? Any way to help? Thanks!

File: types.d.ts (hunk 1 above)
Comment (vilyapilya) [reply types.d.ts]: Yes, please change it to string. We should be able to add a custom level in the options object.

File: types.d.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    /**
+     * A timestamp in ms, must be within one day otherwise it will be dropped and Date.now()
+     * will be used in its place.
+     */
+    timestamp?: number;
+
+    tags?: string | string[];
+  }
+
+  export interface LogDna {
```
Comment (vilyapilya) [inline types.d.ts]: Please rename it to `Logger` since it is describing the Logger obj.

File: types.d.ts (hunk 1 above)
Comment (tjoskar) [reply types.d.ts]: Thanks @vilyapilya, I have removed this type and set the level type to string. See: https://github.com/logdna/nodejs/pull/71/commits/ce78ca83e9cd3701d7794eb12e8324af37eab2c0

File: types.d.ts (hunk 4 above)
Comment (tjoskar) [reply types.d.ts]: Thanks @vilyapilya and I agree, Logger is a much better name.

**Review by vilyapilya — APPROVED** [review]

**Pull request merged**